pkg/cmd/experimental/sync-groups/dataextractor_schema1.go

```diff
@@ -0,0 +1,72 @@
+package group
+
+import "github.com/go-ldap/ldap"
+
+// FirstClassGroupDataExtractor extracts the member list and group metainformation from an LDAP server
+// with first-class LDAP entries for groups
+// ----------FOR SCHEMA 1----------
+type FirstClassGroupDataExtractor struct {
```

Comment:
I think these are going to be worthy of being in a separate package.  It'll make the names shorter in addition to logically separating them.